syft/rekor/queryRekor/utils.go

```diff
@@ -0,0 +1,274 @@
+package queryRekor
+
+import (
+	"bytes"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"regexp"
+
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/file"
+	"github.com/sigstore/cosign/pkg/cosign"
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/spdx/tools-golang/spdx"
+	"github.com/spdx/tools-golang/tvloader"
+)
+
+var trustedCertIdentities = []cosign.Identity{
+	{Subject: "mdeicas@google.com", Issuer: "sigstore.dev"},
+	{Subject: "mdeicas@google.com", Issuer: "sigstore"},
+}
+
+// TODO: identities list is not working. Needs to work to verify github action runs because they don't have an email field
+var certCheckOpts = &cosign.CheckOpts{
+	//Identities: trustedCertIdentities,
+	//CertEmail: "mdeicas@google.com",
+
+	// Why is there an option to prove sigVerifier?
+	// addresses after SAN. what is this and should we check it
+
+}
+
+type spdxSbomWithMetadata struct {
+	sbom        spdx.Document2_2
+	packageName string
+	namespace   string // i.e.  uri
+}
+
+type digest struct {
+	Sha256 string
+}
+
+type subject struct {
+	Name   string
+	Digest digest
+}
+
+type lumjjbSbomType struct {
+	Format string
+	Digest digest
+	Uri    string
+}
+
+type buildMetadata struct {
+	ArtifactSourceRepo             string `json:"artifact-source-repo"`
+	ArtifactSourceRepoCommit       string `json:"artifact-source-repo-commit"`
+	AttestationGeneratorRepo       string `json:"attestation-generator-repo"`
+	AttestationGeneratorRepoCommit string `json:"attestation-generator-repo-commit"`
+}
+
+type lumjjbPredType struct {
+	Sboms         []lumjjbSbomType
+	BuildMetadata buildMetadata `json:"build-metadata"`
+}
+
+type inTotoAttestation struct {
```

Comment:
Reuse of the intoto statement header would be good here. The way they define provenance statement would be similar to how we can define the sbomStatement